apps/end-tenancy/models/upload.js

```diff
@@ -2,69 +2,77 @@
 'use strict';
 
 const url = require('url');
+const FormData = require('form-data');
 const Model = require('hof').model;
 const config = require('../../../config');
 const debug = require('debug')('upload-model');
 
 module.exports = class UploadModel extends Model {
-  save() {
-    return new Promise((resolve, reject) => {
+  async save() {
+    try {
       const attributes = {
         url: config.upload.hostname
       };
       const reqConf = url.parse(this.url(attributes));
-      reqConf.formData = {
-        document: {
-          value: this.get('data'),
-          options: {
-            filename: this.get('name'),
-            contentType: this.get('mimetype')
-          }
-        }
-      };
+      const formData = new FormData();
+      formData.append('document', this.get('data'), {
+        filename: this.get('name'),
+        contentType: this.get('mimetype')
+      });
+      reqConf.data = formData;
       reqConf.method = 'POST';
+      reqConf.headers = {
+        ...formData.getHeaders()
+      };
       debug('SAVE PDF DATA', reqConf);
-      this.request(reqConf, (err, data) => {
-        if (err) {
-          return reject(err);
-        }
-        debug('RESPONSE FROM FILE VAULT SAVE', err, data);
-        return resolve(data);
+      const responseData = await new Promise((resolve, reject) => {
+        this.request(reqConf, (err, data) => {
+          if (err) {
+            return reject(err);
+          }
```

Comment:
```suggestion
      const response = await this.request(reqConf);
      const responseData = response.data;
```
I don't think you need to create a new Promise as axios returns a promise that resolves with a response and it'll be caught in your try, catch

Reply:
👍🏾 I have implemented this now